tests/Microsoft.Bot.Builder.Ai.LUIS.Tests/LuisRecognizerTests.cs

```diff
@@ -410,14 +397,6 @@ private void AssertScore(JToken scoreToken)
             Assert.IsTrue(score <= 1);
         }
 
-        private bool EnvironmentVariablesDefined() => appId != null && subscriptionKey != null && region != null;
```

Comment:
Would have been nice to leave the real recognizer infrastructure.  That way you could replace the mock with the real call to get a new mock or oracle file.  Otherwise to update the tests will be a pain.  Ideally you could create the mock in one place where you could make this change.

Reply:
Actually, the first approach was to maintain a shared set of test logic that can perform real calls to the service or use mocks. But it was [dropped](https://github.com/Microsoft/botbuilder-dotnet/pull/772#pullrequestreview-138859206) in favor of keeping a set of unit tests and another set of [integration tests](https://github.com/Microsoft/botbuilder-dotnet/pull/765) that perform the calls to the real service.

Reply:
But that kind of misses the point.  If the LUIS output changes, someone will have to manually update the mock files and make sure that each of them corresponds to the right queries in the static tests.  It would be much better if you could run against the actual service to update the current static mock files.

Reply:
We found the [Azure.Test.HttpRecorder](https://www.nuget.org/packages/Microsoft.Azure.Test.HttpRecorder) used in the [LUIS Runtime.Tests](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Runtime.Tests/BaseTest.cs#L26-L35) which seems to achieve what you were describing. However, we also found that this NuGet is incompatible with `NETCOREAPP2.0` but they seem to be working on getting a [new version](https://github.com/Azure/azure-sdk-for-net/tree/psSdkJson6/src/SdkCommon/TestFramework/Microsoft.Azure.Test.HttpRecorder) with compatibility. We created issue #826 to keep track of the status of the NuGet and update the LUIS unit tests.

We could also implement the same approach ourselves updating the custom mocks to store/retrieve responses from file system but it sounds like we'll be doing some rework. Thoughts?

Reply:
I love your changes to add the http client listener to LUIS.  Lets get these changes checked in and we can eventually add the ability to run live and see if the mocked files need to change later.